internal/executor/instance/persistentworker/isolation/tart/vm.go

```diff
@@ -36,14 +36,14 @@ func NewVMClonedFrom(
 	env map[string]string,
 	logger *echelon.Logger,
 ) (*VM, error) {
-	subCtx, subCtxCancel := context.WithCancel(ctx)
+	runningVMCtx, runningVMCtxCancel := context.WithCancel(context.Background())
```

Comment:
Is there a need to pass `ctx` now?

Reply:
Yes, it is used multiple times just a couple of lines below.

Reply:
Should we use this context there though?